src/raven_api/indicators.py

```diff
@@ -289,19 +430,36 @@ def calculate_all_indicators(
         parquet_path: Path to Raven-generated Parquet file.
         EFN_threshold: Proportion of MAF used for low flow threshold (e.g., 0.2).
         break_point: Water year to split subperiods (e.g., 2000), or None for full period.
+        sites: List of site IDs to filter (optional).
+        start_date: Start date for filtering data (optional).
+        end_date: End date for filtering data (optional).
 
     Returns:
         DataFrame containing calculated indicators for each site and subperiod.
     """
     try:

```

Comment:
Is there a reason why we need to filter and collect data into a dataframe and convert back to a parquet? Could we not progressively build an output dataframe using all of the indicator methods with the parameters directly?